internal/impl/qdrant/output.go

```diff
@@ -0,0 +1,237 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package qdrant
+
+import (
+	"context"
+	"fmt"
+
+	pb "github.com/qdrant/go-client/qdrant"
+	"github.com/redpanda-data/benthos/v4/public/bloblang"
+	"github.com/redpanda-data/benthos/v4/public/service"
+)
+
+const (
+	qoFieldBatching       = "batching"
+	qoFieldGrpcHost       = "grpc_host"
+	qoFieldAPIToken       = "api_token"
+	qoFieldUseTLS         = "tls"
+	qoFieldCollectionName = "collection_name"
+	qoFieldID             = "id"
+	qoFieldVectorMapping  = "vector_mapping"
+	qoFieldPayloadMapping = "payload_mapping"
+)
+
+func outputSpec() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Version("4.32.2").
+		Categories("AI").
+		Summary("Adds items to a [Qdrant](https://qdrant.tech/) collection").
+		Description(service.OutputPerformanceDocs(true, true)).
+		Fields(
+			service.NewOutputMaxInFlightField(),
+			service.NewBatchPolicyField(qoFieldBatching),
+			service.NewStringField(qoFieldGrpcHost).
+				Description("The gRPC host of the Qdrant collection. Defaults to 'localhost:6334'.").Default("localhost:6334"),
+			service.NewStringField(qoFieldAPIToken).
+				Secret().
+				Description("The Qdrant API token for authentication. Defaults to an empty string.").Default(""),
+			service.NewTLSToggledField(qoFieldUseTLS).Description("TLS(HTTPS) config to use when connecting"),
+			service.NewInterpolatedStringField(qoFieldCollectionName).
+				Description("The name of the collection in Qdrant."),
+			service.NewBloblangField(qoFieldID).
+				Description("The ID of the point to insert. Can be a UUID string or positive integer.").
+				Example(`root = "dc88c126-679f-49f5-ab85-04b77e8c2791"`).
```

Comment:
In terms of ergonomics for fields which need to just resolve to a single value, would it not make more sense to use interpolation here so the users could do `${! uuid_v4() }` or `832`?

Reply:
I tried that. Since it's interpolated _string_ and evaluates to a string in the code, it felt off trying to parse it into a number. Where `bloblang` fields evaluated to `any`.